numpy/lib/tests/test_stride_tricks.py

```diff
@@ -336,6 +336,33 @@ def test_as_strided():
     assert_equal(a.dtype, a_view.dtype)
     assert_array_equal([r] * 3, a_view)
 
+
+def test_sliding_window_view():
```

Comment:
seems like tests with 1/2/3-D arrays at least are needed here.

also a test with non-default strides.

and in the issue there's a comment that this implementation is more numerically stable than the pandas rolling window implementation. would be nice if that could be shown/guaranteed with a test that's numerically challenging

Reply:
What does "numerically stable" mean for an procedure that doesn't perform any arithmetic?

Reply:
It’s certainky more numerically stable to calculate a rolling window mean separately for each window. Pandas uses an algorithm that involves keeping track of a single “current” sum for the rolling window, so every element gets added and then subtracted. Of course, this can be way faster....